sys/include/net/gnrc/netif/lorawan.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup net_gnrc_netif
+ * @{
+ *
+ * @file
+ * @brief  LoRaWAN adaption for @ref net_gnrc_netif
+ *
+ * @author  Jose Ignacio Alamos <jose.alamos@haw-hamburg.de>
+ */
+#ifndef NET_GNRC_NETIF_LORAWAN_H
+#define NET_GNRC_NETIF_LORAWAN_H
+
+#include "net/gnrc/lorawan.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   A Link Check request was scheduled
+ */
+#define GNRC_NETIF_LORAWAN_FLAGS_LINK_CHECK                (0x1U)
+
+/**
+ * @brief   GNRC LoRaWAN interface descriptor
+ */
+typedef struct {
+    uint8_t nwkskey[LORAMAC_NWKSKEY_LEN];   /**< network SKey buffer */
+    uint8_t appskey[LORAMAC_APPSKEY_LEN];   /**< App SKey buffer */
+    uint8_t appkey[LORAMAC_APPKEY_LEN];     /**< App Key buffer */
+    uint8_t deveui[LORAMAC_DEVEUI_LEN];     /**< Device EUI buffer */
+    uint8_t appeui[LORAMAC_APPEUI_LEN];     /**< App EUI buffer */
```

Comment:
this is big, right? Do we need these 3 in memory, IIRC after joining with OTAA using these we have `nwkskey` and `appskey` for that session, so could we save some RAM here?

Reply:
I can move them to the `sc_netif` files.
We need them in RAM for the shell (e.g `ifconfig set appkey`), but I agree we don't need it in RAM for most LoRaWAN applications

Reply:
after second thoughts, I think it will be hard to remove this now.
In the shell files we only have the interface as context. So, if we don't keep the keys there we cannot do `ifconfig set appkey ...` :(

I propose to do this optimization in a follow up, considering the PR still consumes > 1kb less than the semtech pkg

Reply:
To clarify this is more a question bc I'm unsure, so I rely on you expertise. But you seem to agree that these values are not generally required. So I would suggest to move them as you proposed. Because in my lora app I wouldn't use the shell interface to config lora.

Reply:
Okay leave as is, and optimise later.